src/Database/LSMTree/Internal/Run/Index/Compact/Construction.hs

```diff
@@ -66,15 +68,17 @@ import           Database.LSMTree.Internal.Serialise
 -- construction](#incremental).
 data MCompactIndex s = MCompactIndex {
     -- * Core index structure
+    -- | Accumulates the range finder bits 'ciRangeFinder'.
     mciRangeFinder          :: !(VU.MVector s Word32)
   , mciRangeFinderPrecision :: !Int
     -- | Accumulates a chunk of 'ciPrimary'.
   , mciPrimary              :: !(STRef s (VU.MVector s Word32))
-    -- | Accumulates a chunk of 'ciClashes'
-  , mciClashes              :: !(STRef s (VU.MVector s Bit))
+    -- | Accumulates chunks of 'ciClashes'.
+  , mciClashes              :: !(STRef s (NonEmpty (VU.MVector s Bit)))
+    -- | Accumulates the 'ciTieBreaker'.
   , mciTieBreaker           :: !(STRef s (Map SerialisedKey Int))
-    -- | Accumulates a chunk of 'ciLargerThanPage'
-  , mciLargerThanPage       :: !(STRef s (VU.MVector s Bit))
+    -- | Accumulates chunks of 'ciLargerThanPage'.
+  , mciLargerThanPage       :: !(STRef s (NonEmpty (VU.MVector s Bit)))
```

Comment:
I think this is okay as a way of "dynamically growing" the vector. It's somewhat unfortunate that we don't know the length of the `C` and `LTP` vectors initially

Reply:
Yes. Not ideal, but with the right chunk size it should be okay.